config/routes.rb

```diff
@@ -14,12 +14,12 @@
         end
       end
       resources :authors
-      resources :categories
-      resources :tags
+      resources :categories, param: :slug
+      resources :tags, param: :slug
     end
 
-    get '/posts'=> 'posts#index'
-    resources :posts, only: [:index, :show], path: ''
+    get '/posts' => 'posts#index'
+    resources :posts, only: [:index, :show]
```

Comment:
I think this `resources` statement makes the other `/posts` route definitions redundant.